### PR TITLE
Custom Translation Feature - Helper Functions and Classes

### PR DESCRIPTION
**Pull request by JoshuaSchaefer**
Helper functions and classes for the custom translation feature. These will help make
API calls to get, put and delete a new custom translation. In addition, create a swal popup
and show the snackbar with specific text.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
This is the initial PR for the custom translation feature. A few more PRs for this feature will be coming to actually allow the user to enter their custom translation and for a user to see the program owner's custom translation.

### Tests - additional information
There are not any independent tests for the helper functions and classes. There will be tests for this code in future PRs.

### CODE REVIEW

File: assets/js/api/CustomTranslationApi.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import $ from 'jquery'
+
+export class CustomTranslationApi {
+  constructor (programSection) {
+    this.programSection = programSection
+  }
+
+  getCustomTranslation (programId, language, successCallback, errorCallback) {
+    $.ajax({
+      url: '../translate/custom/project/' + programId + '?field=' + this.programSection + '&language=' + language,
```
Comment (danielzhang130) [inline assets/js/api/CustomTranslationApi.js:10]: Could we use router here instead, or does it take a bit too much research to get it to work now?

File: assets/js/custom/SnackbarNotification.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { showSnackbar } from '../components/snackbar'
+
+export class SnackbarNotification {
```
Comment (danielzhang130) [inline assets/js/custom/SnackbarNotification.js]: `js/custom/SnackbarNotification` is a bit too generic. Can it be more specific like the other classes in this commit?
Also `Notification` is a bit redundant

File: assets/js/api/CustomTranslationApi.js (hunk 1 above)
Comment (JoshuaSchaefer) [reply assets/js/api/CustomTranslationApi.js:10]: I think it is okay for now and then we can create another PR that uses the router if we think it is worth doing that

Comment (danielzhang130) [reply assets/js/api/CustomTranslationApi.js:10]: sure

**Review by danielzhang130 — APPROVED** [review]
The PR name could be more specific. The commit message is fine like that

**Pull request closed without merging**